syntax = "proto3";

package secshrnet;

message Share {
    uint32 index = 1;
    bytes key_share = 2;
    bytes ciphertext = 3;
    bytes ciphertext_hash = 4;
}

message Machine {
    string os = 1;
    string name = 2;
    uint64 free = 3;
}

enum PacketType {
    STORE_SHARE = 0;
    ACK_STORE_SHARE = 1;
    RECOVER_SHARE = 2;
    ACK_RECOVER_SHARE = 3;
    NAK_RECOVER_SHARE = 4;
    LIST_TAGS = 5;
    ACK_LIST_TAGS = 6;
    NAK_LIST_TAGS = 7;
    INFO_MACHINE = 8;
    ACK_INFO_MACHINE = 9;
    HEARTBEAT = 10;
}

message Packet {
    PacketType type = 1;
    string sender = 2;
    string tag = 4;
    Share share = 5;
    string hex_tags = 6;
    Machine machine = 7;
}
